edx_solutions_api_integration/users/serializers.py

```diff
@@ -46,6 +46,21 @@ class UserSerializer(DynamicFieldsModelSerializer):
     roles = serializers.SerializerMethodField('get_user_roles')
     grades = serializers.SerializerMethodField('get_user_grades')
     attributes = serializers.SerializerMethodField('get_organization_attributes')
+    course_groups = serializers.SerializerMethodField('get_user_course_groups')
+
+
+    def get_user_course_groups(self, user):
+        """Return a list of course groups of the users, optionally filtered by course id."""
+
+        course_groups = user.course_groups.all()
+
+        if course_groups and 'course_id' in self.context:
```

Comment:
I think checking if `course_groups` is truthy isn't necessary here. If it's falsy but we have `'course_id'` in context, the truth branch here still returns the same result as the false branch, because the list is empty.

Reply:
Additionally we can be fancy and just do

```python
return [
    group.name for group in user.course_groups.all()
    if group.course_id == self.context.get('course_id', group.course_id)
]
```

but this is probably harder to read, so up to you :p

Reply:
@UmanShahzad, your suggestion feels more pythonic but the `self.context.get('course_id', group.course_id)` part is not very obvious after a quick first look.  So I will remove the check for the truthiness of `course_groups`.

Reply:
No worries, looks good now!